app.js

```diff
@@ -12,7 +13,57 @@ app.set('views', __dirname + '/views');
 app.use(express.static(__dirname + '/public'));
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+    clientId: process.env.CLIENT_ID,
+    clientSecret: process.env.CLIENT_SECRET
+  });
+
+  spotifyApi
+  .clientCredentialsGrant()
+  .then(data => spotifyApi.setAccessToken(data.body['access_token']))
+  .catch(error => console.log('Something went wrong when retrieving an access token', error));
 
 // Our routes go here:
+app.get('/', (req, res) => {
+    res.render('home');
+});
+
+
+
+app.get('/artist-search', (req, res) => {
+    spotifyApi
+  .searchArtists(req.query.artistName)
+  .then(data => {
+    console.log('The received data from the API: ', data.body.artists.items[0]);
+    // ----> 'HERE WHAT WE WANT TO DO AFTER RECEIVING THE DATA FROM THE API'
+    res.render('artistSearchResults', {
+        searchedArtistArr: data.body.artists.items
+    })
+  })
+  .catch(err => console.log('The error while searching artists occurred: ', err));
+});
+
+app.get('/albums/:artistId', (req, res) => {
+    spotifyApi
+    .getArtistAlbums(req.params.artistId)
+    .then(albums => {
+        res.render('albums', {
+            albums: albums.body.items
+        })
+    })
```

Comment:
It is a good approach to always add the ``.catch`` when dealing with promises ;)
```
.catch((err) =>
      console.log("The error while searching artists occurred: ", err)
    );
```